tt_metal/impl/buffers/circular_buffer_types.hpp

```diff
@@ -50,29 +50,39 @@ class CircularBufferConfig {
 
     std::optional<uint32_t> globally_allocated_address() const;
 
+    const std::unordered_set<uint8_t> &buffer_indices() const;
+    const std::unordered_set<uint8_t> &local_buffer_indices() const;
+    const std::unordered_set<uint8_t> &remote_buffer_indices() const;
+
     const std::array<std::optional<tt::DataFormat>, NUM_CIRCULAR_BUFFERS> &data_formats() const;
```

Comment:
Just a suggestion here...
Instead of adding these sets you could instead introduce 
```
struct ConfigData {
    tt::DataFormat data_format;
    uint32_t page_size;
    std::optional<Tile> tile;
    bool is_remote;
};
// With only 32 a std::array is nice.  
std::array<ConfigData, NUM_CIRCULAR_BUFFERS> configs_{};
// alternatively
std::array<std::optional<ConfigData>, NUM_CIRCULAR_BUFFERS> configs_{};
```

Reply:
I think it might be better but this was just a suggestion though.  Up to you.